src/hepattn/models/maskformer.py

```diff
@@ -145,8 +160,26 @@ def forward(self, inputs: dict[str, Tensor]) -> dict[str, Tensor]:
             if isinstance(task, ObjectClassificationTask):
                 # Assume that the classification task has only one output
                 x["class_probs"] = outputs["final"][task.name][task.outputs[0]].detach()
-
-        return outputs
+            if isinstance(task, ObjectHitMaskTask) and self.log_task_attn_mask:
+                attn_logits = outputs["final"][task.name][task.outputs[0]].detach()
+                self.output_attn_mask_logging(attn_logits, x)
+        return outputs, self.sorting.sort_indices
+
+    def output_attn_mask_logging(self, attn_logits, x, threshold=0.1):
+        if ((attn_logits is not None) and (self.log_step % N_STEPS_LOG_ATTN_MASK == 0)) or (not self.training):
+            if not hasattr(self, "output_attn_masks_to_log"):
+                self.output_attn_masks_to_log = {}
+            # sigmoid the attn mask to get the probability of the hit being attended to
+            attn_mask_im = attn_logits[0].detach().cpu().clone().sigmoid()
+            self.output_attn_masks_to_log["final"] = {
+                "mask": attn_mask_im >= threshold,
+                "probs": attn_mask_im,
+                "step": self.log_step,
+                "layer": "final",
+            }
```

Comment:
If the `attn_logits` are stored in `outputs`, then I think we should be able to move all fo this to the callback, which has access to `trainer.global_step`, so we can remove the step counter here, and then retrieve the `attn_logits` and turn them into the binary mask in the callback itself